db/versions/0025.yml

```diff
@@ -0,0 +1,278 @@
+          worker_pool_errors.error_id = decode_string_key(wmworker_pool_errors_entities_modify.row_key) and
+          worker_pool_errors.etag = wmworker_pool_errors_entities_modify.old_etag;
+
+        if found then
+          return query select new_row.etag;
+          return;
+        end if;
+
+        perform worker_pool_errors.etag from worker_pool_errors
+        where
+          worker_pool_errors.worker_pool_id = decode_string_key(wmworker_pool_errors_entities_modify.partition_key) and
+          worker_pool_errors.error_id = decode_string_key(wmworker_pool_errors_entities_modify.row_key);
+
+        if found then
+          raise exception 'unsuccessful update' using errcode = 'P0004';
+        else
+          raise exception 'no such row' using errcode = 'P0002';
+        end if;
+      end
+  wmworker_pool_errors_entities_scan:
+    description: See taskcluster-lib-entities
+    mode: read
+    serviceName: worker_manager
+    args: pk text, rk text, condition text, size integer, page integer
+    returns: table (partition_key text, row_key text, value jsonb, version integer, etag uuid)
+    body: |-
+      declare
+      begin
+        if not condition is null then
+          raise exception 'condition not supported';
+        end if;
+        return query
+          select
+            encode_string_key(wmworker_pool_errors_entities_scan.partition_key),
+            encode_string_key(wmworker_pool_errors_entities_scan.row_key),
+            entity_buf_encode(
+              jsonb_build_object(
+                'PartitionKey', encode_string_key(worker_pool_id),
+                'RowKey', encode_string_key(error_id),
+                'errorId', slugid_to_uuid(error_id),
+                'workerPoolId', worker_pool_id,
+                'reported', reported,
+                'kind', kind,
+                'title', title,
+                'description', description),
+              'extra', extra::text) as value,
+            1 as version,
+          worker_pool_errors.etag as etag
+          from worker_pool_errors
+          where
+            partition_key is null or
+            worker_pool_id = decode_string_key(partition_key) and
+            row_key is null or
+            error_id = decode_string_key(row_key)
+          order by error_id
+          limit case
+            when (size is not null and size > 0) then size + 1
+            else null
+          end
+          offset case
+            when (page is not null and page > 0) then page
+            else 0
+          end;
+      end
+  create_worker_pool_error:
+    description: |-
+      Create a new worker pool error.  Raises UNIQUE_VIOLATION if the error already exists.
+    mode: write
+    serviceName: worker_manager
+    args: error_id_in text, worker_pool_id_in text, reported_in timestamptz, kind_in text, title_in text, description_in text, extra_in jsonb
+    returns: uuid
```

Comment:
The etag is not used in the service so probably best to either return nothing (i.e., `void`) or a table set of length 1 and without the etag.

Reply:
Actually, it's fine for now. We can always remove this later :)

Reply:
@imbstack asked a similar question in #3115.  Am I correct that the general approach is to keep etags around where that's not difficult, even if they're not strictly necessary?

Reply:
Yes, keep them around but not return them in the new db functions. Thoughts?

Reply:
Sounds good!

Reply:
#3209